File: Baekjoon/Class2/18.1436 영화감독 숌.java
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        int n, title = 665;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        for(int i = 0; i < n; i++) {
            boolean found = false;
            while(!found) {
                title++;
                int dividend = title, sixCount = 0;
                while(dividend != 0) {
                    if(dividend % 10 == 6) { // 6이 세 번 연속되는 수 찾기
                        sixCount++;
                        if(sixCount == 3)
                            found = true;
                    }
                    else
                        sixCount = 0;
                    dividend /= 10;
                }
            }
        }
        System.out.print(title);
        br.close();
    }
}

/* 문자열 이용
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        int n, count = 1, title = 665;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        while(count <= n) {
            title++;
            if(String.valueOf(title).contains("666"))
                count++;
        }
        System.out.print(title);
        br.close();
    }
}
*/

File: Baekjoon/Class2/18.1436 영화감독 숌.cpp
#include <iostream>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, title = 665;
    cin >> n;

    for (int _ = 0; _ < n; _++) {
        bool found = false;
        while (!found) {
            title++;
            int dividend = title, six_count = 0;
            while (dividend) {
                if (dividend % 10 == 6) { // 6이 세 번 연속되는 수 찾기
                    six_count++;
                    if (six_count == 3)
                        found = true;
                }
                else
                    six_count = 0;
                dividend /= 10;
            }
        }
    }
    cout << title;
    return 0;
}

/* 문자열 이용
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, count = 1, title = 665;
    cin >> n;

    while (count <= n) {
        title++;
        if (to_string(title).find("666") != string::npos)
            count++;
    }
    cout << title;
    return 0;
}
*/